### PR TITLE
Java Search support for JEP 447 Super After Statement

### LINKED ISSUE #2079: [22][search] Search support for JEP 447 Statements before super
Investigate (and testcases definitely) for Search. Not expecting any dom change here.
This is for issue https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2083

### PR DESCRIPTION
**Pull request by subyssurendran666**
fixes #2079 

## What it does
fixes #2079 

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/RunJavaSearchTests.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2000, 2023 IBM Corporation and others.
+ * Copyright (c) 2000, 2024 IBM Corporation and others.
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/RunJavaSearchTests.java:2]: Disclaimer to be added after License

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchSuperAfterStatementTests.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+						X x = new X(100);
+						System.out.println(x.i);
+						X x2 = new X(1);
+						System.out.println(x2.i);
+					}
+				}
+				""";
+		this.workingCopies[0] = getWorkingCopy("/JavaSearchBugs/src/X.java", code);
+		search("super(f, f)", CONSTRUCTOR, REFERENCES);
+		assertSearchResults("src/X.java X(int) [super(f, f);] POTENTIAL_MATCH");
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchSuperAfterStatementTests.java:188]: This is an issue - It should be EXACT_MATCH.
need to investigate

**Review by mpalat — COMMENTED** [review]
Please add  a couple of more scenarios from the batch compiler tests

**Review by mpalat — COMMENTED** [review]
@subyssurendran666 Please wait for https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2148 to get resolved.

**Comment by mpalat** [comment]
> @subyssurendran666 Please wait for #2148 to get resolved.

#2148 is resolved - please change  to setUpJavaProject("JavaSearchBugs", "22") and resubmit

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchSuperAfterStatementTests.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+		if (this.wcOwner == null) {
+			this.wcOwner = new WorkingCopyOwner() {};
+		}
+		return getWorkingCopy(path, source, this.wcOwner);
+	}
+
+	@Override
+	public void setUpSuite() throws Exception {
+		super.setUpSuite();
+		JAVA_PROJECT = setUpJavaProject("JavaSearchBugs", "16");
```
[the remark this answers is not in the document]
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchSuperAfterStatementTests.java]: Done

**Comment by subyssurendran666** [comment]
This should merge to Master after the BETA_JAVA22 release. So should be in draft mode.

**Pull request closed without merging**